src/NHibernate/Impl/FutureBatch.cs

```diff
@@ -73,21 +73,36 @@ private IList GetResults()
 			if (results != null)
 				return results;
 
-			var multiApproach = CreateMultiApproach(isCacheable, cacheRegion);
-			var needTransformer = false;
-			foreach (var query in queries)
+			if (!session.Factory.ConnectionProvider.Driver.SupportsMultipleQueries)
 			{
-				AddTo(multiApproach, query.Query, query.ResultType);
-				if (query.Future?.ExecuteOnEval != null)
-					needTransformer = true;
+				results = new List<object>();
```

Comment:
I would prefer here to use a temporary variable:

1. Collect all results
2. Assign to a "results" field.